### PR TITLE
WIP Flatten batches

### PR DESCRIPTION
**Pull request by bkmoovio**

### CODE REVIEW

File: file.go [hunk 1]
```diff
@@ -894,3 +896,79 @@ func segmentFileBatchAddADVEntry(creditBatch Batcher, debitBatch Batcher, entry
 		debitBatch.AddADVEntry(entry)
 	}
 }
+
+// OptimizeFile optimizes a file by flattening batches with the same BatchHeader data into one batch.
+func (f *File) OptimizeFile() (*File, error) {
```
Comment (adamdecaf) [inline file.go]: Could we call this FlattenBatches? Optimize is probably referencing cost relative to the Fed's prices, yes?

File: file.go [hunk 2]
```diff
@@ -894,3 +896,79 @@ func segmentFileBatchAddADVEntry(creditBatch Batcher, debitBatch Batcher, entry
...
+
+	// Slice of BatchHeaders
+	sbh := make([]string, 0)
+	for _, b := range f.Batches {
+		bh := b.GetHeader().String()
+		sbh = append(sbh, bh[:94])
+	}
+
+	// Remove duplicate BatchHeader entries
+	sbh = removeDuplicateBatchHeaders(sbh)
```
Comment (adamdecaf) [inline file.go:929]: If it's easier I've used a `map[string]interface{}` to accumulate unique values before. You can overwrite based on key.

**Review by adamdecaf — COMMENTED** [review]
(Sorry if you weren't looking for a review yet)

File: file.go (hunk 2 above)
Comment (bkmoovio) [reply file.go:929]: yeah I started with that and then decided to go with a slice

File: file.go (hunk 1 above)
Comment (bkmoovio) [reply file.go]: sure

File: file.go (hunk 2 above)
Comment (bkmoovio) [reply file.go:929]: I created code to use a map.   Both are out there to see the difference

**Review by wadearnold — APPROVED** [review]

**Pull request merged**